gamification-evm-services/src/main/java/io/meeds/evm/gamification/scheduling/task/ERC20TransferTask.java

```diff
@@ -37,108 +40,158 @@
 import org.exoplatform.commons.api.settings.data.Scope;
 import org.exoplatform.commons.api.settings.SettingValue;
 
+import org.exoplatform.services.listener.ListenerService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
+import static io.meeds.evm.gamification.utils.Utils.EVM_SAVE_ACTION_EVENT;
+
 @Component
 public class ERC20TransferTask {
-  private static final Logger  LOG                         = LoggerFactory.getLogger(ERC20TransferTask.class);
+  private static final Logger   LOG                         = LoggerFactory.getLogger(ERC20TransferTask.class);
+
+  private static final Scope    SETTING_SCOPE               = Scope.APPLICATION.id("GAMIFICATION_EVM");
+
+  private static final Context  SETTING_CONTEXT             = Context.GLOBAL.id("GAMIFICATION_EVM");
 
-  private static final Scope   SETTING_SCOPE               = Scope.APPLICATION.id("GAMIFICATION_EVM");
+  private static final String   SETTING_LAST_TIME_CHECK_KEY = "transferredTokenTransactionsCheck";
 
-  private static final Context SETTING_CONTEXT             = Context.GLOBAL.id("GAMIFICATION_EVM");
+  @Autowired
+  private SettingService        settingService;
 
-  private static final String  SETTING_LAST_TIME_CHECK_KEY = "transferredTokenTransactionsCheck";
+  @Autowired
+  private BlockchainService     blockchainService;
 
   @Autowired
-  private SettingService       settingService;
+  private EvmTriggerService     evmTriggerService;
 
   @Autowired
-  private BlockchainService    blockchainService;
+  private RuleService           ruleService;
 
   @Autowired
-  private EvmTriggerService    evmTriggerService;
+  private EvmTransactionService evmTransactionService;
 
   @Autowired
-  private RuleService          ruleService;
+  private ListenerService       listenerService;
 
   @ContainerTransactional
   @Scheduled(cron = "0 * * * * *")
   public synchronized void listenTokenTransfer() {
```

Comment:
Can you please transfer all business logic inside a `Service` layer instead of embedding it inside a `Scheduled Task` (you may have to split this into multiple methods)